tools/crldps/main.go

```diff
@@ -0,0 +1,116 @@
+// crldps generates the list of CRL Distribution Point URIs for a given issuing
+// CA and number of shards. It fetches and validates all of those shards. If
+// it doesn't encounter any errors, it pretty-prints the list of all CRLDPs for
+// disclosure in CCADB.
+package main
+
+import (
+	"crypto/x509"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/crl/idp"
+)
+
+func main() {
+	log.SetOutput(os.Stderr)
+
+	fs := flag.NewFlagSet("crldps", flag.ContinueOnError)
+	caPath := fs.String("ca", "", "path to an issuing intermediate CA certificate (required)")
+	numShards := fs.Int("shards", 128, "number of CRL shards issued by the CA")
+	err := fs.Parse(os.Args[1:])
+	if err != nil || len(fs.Args()) != 0 {
+		log.Println("Incorrect command line flags; usage:")
+		fs.PrintDefaults()
+		os.Exit(1)
+	}
+
+	issuer, err := core.LoadCert(*caPath)
+	if err != nil {
+		log.Fatalf("Failed to load issuer certificate from %q: %s", os.Args[1], err)
+	}
+
+	crldpBase := fmt.Sprintf("http://%s.c.lencr.org", strings.ToLower(issuer.Subject.CommonName))
+
+	var (
+		anyErr bool
+		crldps []string
+	)
+	for shard := range *numShards {
+		// We 1-index our CRL shards.
+		crldp := fmt.Sprintf("%s/%d.crl", crldpBase, shard+1)
+
+		resp, err := http.Get(crldp)
+		if err != nil {
+			anyErr = true
+			log.Printf("Error downloading crl %q: %s", crldp, err)
```

Comment:
I can see this `anyErr` bookkeeping going awry in future updates, instead consider replacing all of these with a helper that looks like this:

```go
func logErr(anyErr *bool, format string, args ...any) {
	*anyErr = true
	log.Printf(format, args...)
}
```

Reply:
I had the same thought, but this approach comes with its own downside: the go compiler and linter can't apply their understanding of format verbs to the logErr function. This would make it possible to accidentally write `logErr("oops: %s", crldp, err)` and not have the mismatch caught until runtime.

I think it's fine for now, and if we want to change it the next time we're touching this code, we can tackle it then.